packages/connect/src/SmoldotProvider/SmoldotProvider.test.ts

```diff
@@ -19,9 +17,63 @@ import { jest } from '@jest/globals';
 
 const EMPTY_CHAIN_SPEC = '{}';
 
+describe('Test Development chain', () => {
+  let provider: SmoldotProvider;
+
+  afterEach(async () => {
+    await provider.disconnect();
+  });
+
+  test('emits events when it connects / disconnects / reconnects', async () => {
+    const healthResponses = [
+      { isSyncing: false, peerCount: 1, shouldHavePeers: true },
+      { isSyncing: false, peerCount: 0, shouldHavePeers: true },
+      { isSyncing: false, peerCount: 1, shouldHavePeers: true }
+    ];
+    const ms = mockSmoldot(respondWith([]), customHealthResponder(healthResponses));
+    provider = new SmoldotProvider(EMPTY_CHAIN_SPEC, ms);
+  
+    // we don't want the test to be slow
+    provider.healthPingerInterval = 1;
+    await provider.connect();
+  
+    return new Promise<void>((resolve) => {
+      provider.on('connected', () => {
+        const off = provider.on('disconnected', () => {
+          off(); // stop listening
+          provider.on('connected', () => {
+            provider.disconnect().then(() => resolve());
+          });
+        });
+      });
+    });
+  });
+
+  test('emits connect and never emits disconnect for development chain', async () => {  
```

Comment:
I thought you were going to delete this test

Reply:
As soon as there is a fix that makes it work - why to remove it? It was written to test some functionality - I was suggesting deletion if it was overkill to fix - but as soon as same solution applies for both tests, I don't find a reason to delete it